src/test/ui/issues/issue-73541.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0426]: use of undeclared label `'a`
+  --> $DIR/issue-73541.rs:4:29
+   |
+LL |             loop { continue 'a }
+   |                             ^^
+   |
+   = note: labels cannot be used through closures or async blocks
+   = help: a label with this name exists in a parent scope
```

Comment:
(Haven't checked the code to see if this is possible.)

I would aim at having something closer to

```
error[E0426]: use of undeclared label `'a`
  --> $DIR/issue-73541.rs:4:29
   |
LL | fn main() {
LL |     'a: loop {
   |     -- a label with this name exists in a parent scope but is unreachable
LL |         || {
   |         -- labels cannot be used through closures or async blocks
LL |             loop { continue 'a }
   |                             ^^ undeclared label
```

Reply:
I think I’d prefer this happen in another PR if it were possible, just so that it doesn’t delay landing the ICE fix here (really I should have waited to make the diagnostic change in another PR, but I couldn’t help myself).

Reply:
That's fine by me :)

The current output is good, just not perfect ^_^

Reply:
I've improved the error messages further in the most recent commit - I've not added highlighting of the closure/fn that results in a unreachable label because that wasn't trivial, but I've added other labels.